DEGG PMT matching

BeritSchlueter opened the pull request:
- Adding DEGG PMT matching 
- Changing typo in the in the scan functions
- Changing on line in OMSim Hit that stricter compiler can run the code

Commit:
Adding DEGG PMT matching, correcting a typo in the scan functions, rewriting one line in OMSimHitManager

MarkusDittmer commented on common/framework/src/OMSimHitManager.cc:
What was this change for again?

Did you check e.g. a single effective area direction if all other modules (especially DOM) still behave normally?

BeritSchlueter replied:
I tested it with the mDOM and the D-Egg. The change was needed because else my compiler was complaining. I can test the DOM.,

MarkusDittmer commented on common/data/PMTs/measurement_matching_data/setup_stuff/DEGG_used_z_compensation.txt:
is the need/reason for the change of measurement file documented somewhere?

BeritSchlueter replied:
I needed to change it for step 4, because else the beam started in the PMT glass.

MarkusDittmer replied:
I know, I mean if that is written down (documented) anywhere - e.g. documentation/notebooks/detection_efficiency_matching or some other suitable place. Also add a backup of the original file (from the measurement) to that, else it will be lost over time.

BeritSchlueter replied:
I uploaded the old file and it is written down in the commit of this file and also in the notebook.

MarkusDittmer replied:
Looks reasonable to me. Please wait for [this](https://github.com/icecube/OMSim/pull/20) merge to go through first in case of conflicts.

MarkusDittmer replied:
Previous merge is done which included the first batch of values you gave me. Since you did a few iterations they conflict at this point. Maybe to a quick side by side / plot comparison if some of those values drastically changed, but you already said that the effective area looks fine right?

BeritSchlueter replied:
Sry for the late response. The notification email got in my spam folder. The effective area looks fine.

jvaracarbonell replied:
@BeritSchlueter can you resolve the conflicts?

BeritSchlueter committed:
Adding old DEGG setup_stuff.

This file had to be revised because the laser beam of the simulated scan started in the PMT and not outside the PMT.
New file is DEGG_used_z_compensation.txt

Commit:
Adding comments to the notebook.

jvaracarbonell commented:
@BeritSchlueter could you resolve the conflicts when you find some time?
